src/Symfony/Component/Serializer/Type/Type.php

```diff
@@ -0,0 +1,588 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Type;
+
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+use Symfony\Component\Serializer\Exception\LogicException;
+use Symfony\Component\Serializer\Exception\UnsupportedException;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @experimental in 7.0
+ */
+final class Type implements \Stringable
```

Comment:
Why not improving current `Type` class ?

Reply:
Is it because the current `Type` class lives in the `PropertyInfo` component and so is IMHO associated to a property. 

Therefore, it was strange to me to ask users to write something like:
```php
$this->deserializer->deserialize('...', Type::int(), 'json');
```
with a `Type` class coming from the property info component.

Reply:
Maybe should we extract this part from PropertyInfo in a new tiny low level “Type” or “Reflection” component and make it a dependency of PropertyInfo and Serializer.

Reply:
Indeed, we began to work on that with @Korbeil.